### ISSUE TITLE
Text Component ellipsis is not displayed correctly in Android using custom font (the three dots are cut-off)

### ISSUE THREAD

**Issue by alexxxwork** [facebook/react-native#36350]
### Description

This is a duplicate report of this old bug https://github.com/facebook/react-native/issues/33487 on a current version of RN

When I use custom font and set numberOfLines={1} on a Text component, if the parent's component width is fixed, sometimes the ellipsis is cut-off strangely. For example, in the below screenshot, you can see that only 1 and a half dots of the ellipsis are shown.

It seems to only happen on Android

![Screenshot_20230302_104027_testApp](https://user-images.githubusercontent.com/3872759/222355787-264615d9-230b-4946-866f-7301aa9abc83.jpg)

## Expo Go:

![Screenshot_20230302_114853_Expo Go](https://user-images.githubusercontent.com/3872759/222355833-817fd07e-1fb8-4bb4-823e-1999cd53906c.jpg)

### React Native Version

0.71.3

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Memory: 550.20 MB / 7.85 GB
  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: 20220918.223204.0 - C:\ProgramData\chocolatey\bin\watchman.EXE
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Disabled
      AllowAllTrustedApps: Disabled
  IDEs:
    Android Studio: AI-221.6008.13.2211.9514443
    Visual Studio: 15.7.27703.2026 (Visual Studio Community 2017)
  Languages:
    Java: 17.0.6 - C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0
    react-native: 0.71.3 => 0.71.3
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

add custom font assets and run a minimum code example

### Snack, code example, screenshot, or link to a repository

[Expo link](https://snack.expo.dev/@alexxxwork/text-component-ellipsis-is-not-displayed-correctly?platform=android)
It doesn't reproduce on web android emultator but reproduces when running on physical device through expo.
Font file:
[ExpensifyNewKansas-Medium.zip](https://github.com/facebook/react-native/files/10868503/ExpensifyNewKansas-Medium.zip)

```js
import React from 'react';
import {Text, View} from 'react-native';

function App(): JSX.Element {
  return (
    <View
      style={{
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%',
      }}>
      <View style={{width: 173, padding: 24, backgroundColor: 'blue'}}>
        <Text
          style={{fontSize: 30, fontFamily: 'ExpensifyNewKansas-Medium'}}
          numberOfLines={1}>
          This text will be cut-off strangely in Android
        </Text>
      </View>
    </View>
  );
}

export default App;
```

**Comment by alexxxwork** [comment]
Recreated https://github.com/facebook/react-native/issues/36336 with a minimum code example and latest RN version

**Comment by BeeMargarida** [comment]
I'm investigating this issue and can only replicate it in my Xiaomi device (can't replicate in any emulator - even using the same Android version - nor in any other physical device I have around).

I started by replicating the same scenario in a clean Android app, since it could be an Android bug and not a RN one, and there are 2 bugs reported in the Android issue tracker that are similar to this one([1](https://issuetracker.google.com/issues/273413221), [2](https://issuetracker.google.com/issues/275580756)).
In the Android project I can't replicate the bug, so I'll keep investigating on the RN side. However, I'm not totally sure if my replication is totally 1:1, so if anyone wants to take a look and give their opinion, I would greatly appreciative it. I'll keep investigating.

[TestEllipsis.zip](https://github.com/facebook/react-native/files/11160656/TestEllipsis.zip)

**Comment by alexxxwork** [comment]
@BeeMargarida Hi! Have you tried to launch this expo [link ](https://snack.expo.dev/@alexxxwork/text-component-ellipsis-is-not-displayed-correctly?platform=android) on your device? 
I don't think these Android bugs are related. This [1](https://issuetracker.google.com/issues/273413221) is a WearOS bug. And this [2](https://issuetracker.google.com/issues/275580756) is about line height and not ellipsis cut-off

**Comment by BeeMargarida** [comment]
> @BeeMargarida Hi! Have you tried to launch this expo [link ](https://snack.expo.dev/@alexxxwork/text-component-ellipsis-is-not-displayed-correctly?platform=android) on your device?

Yap, can only replicate the bug on my Xiaomi device

>  I don't think these Android bugs are related. This [1](https://issuetracker.google.com/issues/273413221) is a WearOS bug. And this [2](https://issuetracker.google.com/issues/275580756) is about line height and not ellipsis cut-off

Agree, but the root cause can be the same

**Comment by BeeMargarida** [comment]
Could replicate in emulator using default font - https://snack.expo.dev/@daisy112351/text-component-ellipsis-is-not-displayed-correctly?platform=android

<img width="331" alt="image" src="https://user-images.githubusercontent.com/25725586/230630291-81b8818d-241f-4fc9-b817-13528c6ba114.png">

**Comment by MatiPl01** [comment]
Why is this issue closed and the PR hasn't been merged? Is the issue resolved? I still have the same problem in RN 0.71.11.

**Comment by mleister97** [comment]
Any updates on this one?
